tests/end-to-end/external-solution-integration/internal/scenario/event_mesh/event_mesh.go

```diff
@@ -81,7 +89,6 @@ func (s *Scenario) Steps(config *rest.Config) ([]step.Step, error) {
 		testsuite.NewCreateServiceBindingUsage(s.testID, s.testID, s.testID,
 			serviceBindingUsageClientset.ServicecatalogV1alpha1().ServiceBindingUsages(s.testID),
 			knativeEventingClientSet.EventingV1alpha1().Brokers(s.testID), knativeEventingClientSet.MessagingV1alpha1().Subscriptions(kymaIntegrationNamespace)),
-		testsuite.NewSleep(s.waitTime),
 		testsuite.NewCreateKnativeTrigger(s.testID, defaultBrokerName, functionEndpoint, knativeEventingClientSet.EventingV1alpha1().Triggers(s.testID)),
 		testsuite.NewSleep(s.waitTime),
```

Comment:
I see `testsuite.NewCreateKnativeTrigger` is also being used in `compass.go`, `connectivity_adapter.go`. I think finally they should look similiar regarding sleeps/retry for eventing.
Also `event_mesh.go` uses `NewWrappedCounterPod` in contrast to `testsuite.NewCheckCounterPod` in the other two mentioned files. I think we should have the same retry behaviour across all these tests 😎

Reply:
Without the 10s sleep after "testsuite.NewCreateKnativeTrigger" the test event is somewhere lost. I suppose that it's related to our Kafka integration to Event-Hub, as discussed earlier. With the current implementation, a simple test could have another retry behaviour compared to e2e test, which allows a  better simulation of the reality. The e2e test is the most complex test till now. Each test could define it's specific retry behaviour on different calling sequences as needed. This will overwrite the lower "retry" level and at the last end of the calling sequence the defaults values defined in "github.com/avast/retry-go" (100ms, etc.).

Reply:
is there any state that we could check for before we send an event?
just sleeping 'long enough' does not really sound as if the issue is understood. Do we know why we need to wait?

Reply:
From the API perspective, if a trigger is marked as "ready", like in our case, the caller should be able to use it. It seems that we have found an "edge" case for the Event-Hub integration, where without waiting 10s, the event is not sent to Event-Hub, in spite of the "202" returned by istio-proxy (the side-car of kantive-kafka-channel). See also the comments regarding Event-Hub messaging in: https://github.com/kyma-project/kyma/issues/8283